src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Search/SearchIndexEntryJObjectGenerator.cs

```diff
@@ -83,57 +91,83 @@ void ISearchValueVisitor.Visit(NumberSearchValue number)
             {
                 AddProperty(SearchValueConstants.NumberName, number.Low);
             }
-
-            AddProperty(SearchValueConstants.LowNumberName, number.Low);
-            AddProperty(SearchValueConstants.HighNumberName, number.High);
+            else
+            {
+                AddProperty(SearchValueConstants.LowNumberName, number.Low);
+                AddProperty(SearchValueConstants.HighNumberName, number.High);
+            }
         }
 
         void ISearchValueVisitor.Visit(QuantitySearchValue quantity)
         {
-            AddPropertyIfNotNull(SearchValueConstants.SystemName, quantity.System);
+            ////AddPropertyIfNotNull(SearchValueConstants.SystemName, quantity.System);
+            AddPropertyIfNotNullWithIntMap(SearchValueConstants.SystemName, quantity.System);
             AddPropertyIfNotNull(SearchValueConstants.CodeName, quantity.Code);
 
             if (quantity.Low == quantity.High)
             {
                 AddProperty(SearchValueConstants.QuantityName, quantity.Low);
             }
-
-            AddProperty(SearchValueConstants.LowQuantityName, quantity.Low);
-            AddProperty(SearchValueConstants.HighQuantityName, quantity.High);
+            else
+            {
+                AddProperty(SearchValueConstants.LowQuantityName, quantity.Low);
+                AddProperty(SearchValueConstants.HighQuantityName, quantity.High);
+            }
         }
 
         void ISearchValueVisitor.Visit(ReferenceSearchValue reference)
         {
             AddPropertyIfNotNull(SearchValueConstants.ReferenceBaseUriName, reference.BaseUri?.ToString());
-            AddPropertyIfNotNull(SearchValueConstants.ReferenceResourceTypeName, reference.ResourceType?.ToString());
+            ////AddPropertyIfNotNull(SearchValueConstants.ReferenceResourceTypeName, reference.ResourceType?.ToString());
```

Comment:
Same question here as I'm not sure of the context of this change. There are several items commented out. Shouldn't they just be removed?